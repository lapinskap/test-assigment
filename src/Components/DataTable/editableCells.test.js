import renderer from 'react-test-renderer';
import {
  getEditableCell,
  getToggleSwitchCell,
  getEditableSelectCell,
  getEditableCheckboxesCell,
} from './editableCells';
import '../../tests/setupTests';

describe('<getEditableCell />', () => {
  let componentRenderer; let data; let updateData; const type = 'text'; let disableDependsOn = null; let
    cellInfo;

  beforeEach(() => {
    data = [
      {
        randomField: 'Random value',
      },
    ];
    updateData = () => ({});
    cellInfo = {
      index: 0,
      column: {
        id: 'randomField',
      },
    };
    componentRenderer = getEditableCell(data, updateData, type, disableDependsOn);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with props', () => {
    const tree = renderer
      .create(componentRenderer(cellInfo))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with disableDependsOn prop', () => {
    disableDependsOn = true;
    const tree = renderer
      .create(componentRenderer(cellInfo))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<getToggleSwitchCell />', () => {
  let componentRenderer; let data; let updateData; const type = 'text'; const disableDependsOn = null; let
    cellInfo;

  beforeEach(() => {
    data = [
      {
        style: undefined,
        id: 1,
      },
    ];
    updateData = () => ({});
    cellInfo = {
      index: 0,
      column: {
        id: 1,
      },
    };
    componentRenderer = getToggleSwitchCell(data, updateData, type, disableDependsOn);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with props', () => {
    const tree = renderer
      .create(componentRenderer(cellInfo))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<getEditableSelectCell />', () => {
  let componentRenderer; let data; let updateData; let options = []; let
    cellInfo;

  beforeEach(() => {
    data = [
      { isEnabled: 1 },
    ];
    updateData = () => ({});
    cellInfo = {
      index: 0,
      column: {
        id: 'isEnabled',
      },
    };
    options = [
      { value: 0, label: 'Nie' },
      { value: 1, label: 'Tak' },
    ];
    componentRenderer = getEditableSelectCell(data, updateData, options);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with props', () => {
    const tree = renderer
      .create(componentRenderer(cellInfo))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<getEditableCheckboxesCell />', () => {
  let componentRenderer; let data; let updateData; let options = []; let
    cellInfo;

  beforeEach(() => {
    data = [
      { features: ['option_1', 'option_2'] },
    ];
    updateData = () => ({});
    cellInfo = {
      index: 0,
      column: {
        id: 'features',
      },
    };
    options = [
      { value: 'option_1', label: 'Opcja 1' },
      { value: 'option_2', label: 'Opcja 2' },
      { value: 'option_3', label: 'Opcja 3' },
    ];
    componentRenderer = getEditableCheckboxesCell(data, updateData, options);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with props', () => {
    const tree = renderer
      .create(componentRenderer(cellInfo))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without data prop', () => {
    data = [];
    const tree = renderer
      .create(componentRenderer(cellInfo))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
