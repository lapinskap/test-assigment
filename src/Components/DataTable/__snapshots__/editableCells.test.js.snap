// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<getEditableCell /> renders correctly with disableDependsOn prop 1`] = `
<div
  className="d-block w-100 text-center"
>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      defaultValue="Random value"
      disabled={false}
      onBlur={[Function]}
      type="text"
    />
  </div>
</div>
`;

exports[`<getEditableCell /> renders correctly with props 1`] = `
<div
  className="d-block w-100 text-center"
>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      defaultValue="Random value"
      disabled={false}
      onBlur={[Function]}
      type="text"
    />
  </div>
</div>
`;

exports[`<getEditableCheckboxesCell /> renders correctly with props 1`] = `
<div
  className="text-left"
>
  <div
    className="custom-checkbox custom-control"
    hidden={false}
  >
    <input
      checked={true}
      className="custom-control-input"
      id="option_1_0"
      onChange={[Function]}
      type="checkbox"
      value="option_1"
    />
    <label
      className="custom-control-label"
      htmlFor="option_1_0"
    >
      Opcja 1
    </label>
  </div>
  <div
    className="custom-checkbox custom-control"
    hidden={false}
  >
    <input
      checked={true}
      className="custom-control-input"
      id="option_2_0"
      onChange={[Function]}
      type="checkbox"
      value="option_2"
    />
    <label
      className="custom-control-label"
      htmlFor="option_2_0"
    >
      Opcja 2
    </label>
  </div>
  <div
    className="custom-checkbox custom-control"
    hidden={false}
  >
    <input
      checked={false}
      className="custom-control-input"
      id="option_3_0"
      onChange={[Function]}
      type="checkbox"
      value="option_3"
    />
    <label
      className="custom-control-label"
      htmlFor="option_3_0"
    >
      Opcja 3
    </label>
  </div>
</div>
`;

exports[`<getEditableCheckboxesCell /> renders correctly without data prop 1`] = `
<div
  className="text-left"
>
  <div
    className="custom-checkbox custom-control"
    hidden={false}
  >
    <input
      checked={true}
      className="custom-control-input"
      id="option_1_0"
      onChange={[Function]}
      type="checkbox"
      value="option_1"
    />
    <label
      className="custom-control-label"
      htmlFor="option_1_0"
    >
      Opcja 1
    </label>
  </div>
  <div
    className="custom-checkbox custom-control"
    hidden={false}
  >
    <input
      checked={true}
      className="custom-control-input"
      id="option_2_0"
      onChange={[Function]}
      type="checkbox"
      value="option_2"
    />
    <label
      className="custom-control-label"
      htmlFor="option_2_0"
    >
      Opcja 2
    </label>
  </div>
  <div
    className="custom-checkbox custom-control"
    hidden={false}
  >
    <input
      checked={false}
      className="custom-control-input"
      id="option_3_0"
      onChange={[Function]}
      type="checkbox"
      value="option_3"
    />
    <label
      className="custom-control-label"
      htmlFor="option_3_0"
    >
      Opcja 3
    </label>
  </div>
</div>
`;

exports[`<getEditableSelectCell /> renders correctly with props 1`] = `
<select
  className="form-control"
  id="isEnabled"
  name="isEnabled"
  onChange={[Function]}
  value={1}
>
  <option
    value={0}
  >
    Nie
  </option>
  <option
    value={1}
  >
    Tak
  </option>
</select>
`;

exports[`<getToggleSwitchCell /> renders correctly with props 1`] = `
<div
  className="d-block w-100 text-center"
>
  <div
    className="mr-2 mb-2"
    style={
      Object {
        "MozTransition": "opacity 0.25s",
        "MozUserSelect": "none",
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "WebkitTransition": "opacity 0.25s",
        "WebkitUserSelect": "none",
        "borderRadius": 10,
        "display": "inline-block",
        "msUserSelect": "none",
        "opacity": 1,
        "position": "relative",
        "textAlign": "left",
        "touchAction": "none",
        "transition": "opacity 0.25s",
        "userSelect": "none",
      }
    }
  >
    <div
      className="react-switch-bg"
      onClick={[Function]}
      onMouseDown={[Function]}
      style={
        Object {
          "MozTransition": "background 0.25s",
          "WebkitTransition": "background 0.25s",
          "background": "#888888",
          "borderRadius": 10,
          "cursor": "pointer",
          "height": 20,
          "margin": 5,
          "position": "relative",
          "transition": "background 0.25s",
          "width": 48,
        }
      }
    />
    <div
      className="react-switch-handle"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
          "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
          "background": "#ffffff",
          "border": 0,
          "borderRadius": "50%",
          "boxShadow": "0px 1px 5px rgba(0, 0, 0, 0.6)",
          "cursor": "pointer",
          "display": "inline-block",
          "height": 30,
          "outline": 0,
          "position": "absolute",
          "top": 0,
          "transform": "translateX(0px)",
          "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
          "width": 30,
        }
      }
    />
    <input
      checked={false}
      disabled={false}
      id="material-switch"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      role="switch"
      style={
        Object {
          "border": 0,
          "clip": "rect(0 0 0 0)",
          "height": 1,
          "margin": -1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "width": 1,
        }
      }
      type="checkbox"
    />
  </div>
</div>
`;
